packages/cozy-doctypes/src/File.js

```diff
@@ -171,12 +170,16 @@ class CozyFile extends Document {
    * @param {ArrayBuffer} file data
    * @param {String} dirId dir id where to upload
    * @param {String} conflictStrategy Actually only 2 hardcoded strategies 'erase' or 'rename'
+   * @param {Object} metadata An object containing the wanted metadata to attach
+   * @param {String} contentType content type of the file
    */
   static async uploadFileWithConflictStrategy(
     name,
     file,
     dirId,
-    conflictStrategy
+    conflictStrategy,
+    metadata,
+    contentType
```

Comment:
With 6 arguments it starts to be hard to remember the right order of argument. Have you considered using option objects ? For example the file options could be an object and the conflict strategy could be a second argument. I think it would also be easier when passing the file options down to other functions.

Reply:
Yep, this is planned when migrating to cozy-client (I write a todo there : https://github.com/cozy/cozy-client/pull/571#issue-344568157). I don't want to make a Breaking Change right now